source-container-build/app/source_build.py

```diff
@@ -331,6 +340,15 @@ def _find_prefetch_source_archives():
         shutil.copy(src, dest)
         sib_dirs.extra_src_dirs.append(f"{prepared_sources_dir}/{src_dir}")
 
+    for root, filename in _find_prefetch_srpm_archives():
+        src_dir = f"src-{next(source_counter)}"
+        copy_dest_dir = f"{prepared_sources_dir}/{src_dir}/{relative_to(root, cachi2_output_dir)}"
+        os.makedirs(copy_dest_dir)
+        src = f"{root}/{filename}"
+        dest = sib_dirs.rpm_dir
+        log.debug("copy prefetched rpm source %s to %s", src, dest)
+        shutil.copy(src, dest)
```

Comment:
IIRC, per the cachi2 design, all RPMs including SRPMs will be stored in architecture-specific directories, which means the same SRPM will likely exist in multiple paths.

Here, we de-duplicate implicitly (by filename). I'm wondering if we need a stronger check, e.g. check that SRPMs with the same filename are really byte-for-byte identical. If they're not, we essentially have three options:
* keep only one of them, document this as a known corner case
* fail the source build, consider this unsupported
* rename one of them and keep both

Reply:
See this thread in the design: https://github.com/containerbuildsystem/cachi2/discussions/465#discussioncomment-8492211

In particular:

> (Source RPMs are not necessarily arch independent, you could have Sources or Patches conditional on architecture. Fedora prohibits that by policy.)

Reply:
Let's do it :+1: